iis-wf/iis-wf-import/src/main/java/eu/dnetlib/iis/wf/importer/infospace/converter/DocumentMetadataConverter.java

```diff
@@ -0,0 +1,318 @@
+    public DocumentMetadata convert(OafEntity oafEntity, Map<String, List<QualifiedOafJsonRecord>> relations) throws IOException {
+        Preconditions.checkNotNull(oafEntity);
+        ResultProtos.Result sourceResult = oafEntity.getResult();
+        if (sourceResult == null) {
+            log.error("skipping: no result object for id " + oafEntity.getId());
+            return null;
+        }
+        DocumentMetadata.Builder builder = DocumentMetadata.newBuilder();
+        builder.setId(oafEntity.getId());
+        createBasicMetadata(sourceResult, builder);
+        handleAdditionalIds(oafEntity, builder);
+        handleDatasourceIds(oafEntity, builder);
+        List<QualifiedOafJsonRecord> personRelations = relations.get(personResultColumnFamily);
+        if (!CollectionUtils.isEmpty(personRelations)) {
+            handlePersons(personRelations, builder);    
+        }
+        return builder.build();
+    }
+
+    // ------------------------ PRIVATE --------------------------
+    
+    /**
+     * Creates basic metadata object.
+     * 
+     * @param sourceResult
+     * @param metaBuilder
+     * @return basic metadata object
+     */
+    private DocumentMetadata.Builder createBasicMetadata(ResultProtos.Result sourceResult,
+            DocumentMetadata.Builder metaBuilder) {
+        if (sourceResult.getMetadata() == null) {
+            return metaBuilder;
+        }
+        handleTitle(sourceResult.getMetadata().getTitleList(), metaBuilder);
+        handleDescription(sourceResult.getMetadata().getDescriptionList(), metaBuilder);
+        handleLanguage(sourceResult.getMetadata().getLanguage(), metaBuilder);
+        handlePublisher(sourceResult.getMetadata().getPublisher(), metaBuilder);
+        handleJournal(sourceResult.getMetadata().getJournal(), metaBuilder);
+        handleYear(sourceResult.getMetadata().getDateofacceptance(), metaBuilder);
+        handleKeywords(sourceResult.getMetadata().getSubjectList(), metaBuilder);
+        handlePublicationType(sourceResult.getInstanceList(), metaBuilder);
+        return metaBuilder;
+    }
+
+    private void handleTitle(List<StructuredProperty> titleList, DocumentMetadata.Builder metaBuilder) {
+        if (!CollectionUtils.isEmpty(titleList)) {
+            for (StructuredProperty titleProp : titleList) {
+                if (titleProp.getQualifier() != null && HBaseConstants.SEMANTIC_CLASS_MAIN_TITLE.equals(titleProp.getQualifier().getClassid())
+                        && fieldApprover.approve(titleProp.getDataInfo())) {
+                    metaBuilder.setTitle(titleProp.getValue());
+                }
+            }
+            if (!metaBuilder.hasTitle()) {
+                // if no main title available, setting first applicable title from the list
+                for (StructuredProperty titleProp : titleList) {
+                    if (fieldApprover.approve(titleProp.getDataInfo())) {
+                        metaBuilder.setTitle(titleProp.getValue());
```

Comment:
The comment says "first" but the whole list is processed, so it is actually "last", isn't it?